cdap-watchdog/src/main/java/co/cask/cdap/logging/pipeline/kafka/KafkaLogProcessorPipeline.java

```diff
@@ -283,23 +285,23 @@ private long processMessages(String topic, int partition,
         // Although it's not the same as the in memory object size, it should be just a constant factor, hence
         // it is proportional to the actual object size.
         eventQueue.add(loggingEvent, loggingEvent.getTimeStamp(),
-                       message.message().payloadSize(), partition, message.offset());
+                       message.message().payloadSize(), partition, message.nextOffset());
```

Comment:
There will be some refactoring needed. The idea is I want to keep the `TimeEventQueue` class neutral and it's up to the caller to use it. I think it's ok for the searching logic to use `offset - 1`.